File: JavaScript/Endpoint Examples/JSON Payload/validated-zugferd.js
// Upload a hybrid PDF, then validate its ZUGFeRD / Factur-X package by resource ID.
const axios = require("axios");
const fs = require("fs");
const path = require("path");

const apiUrl = "https://api.pdfrest.com";
// const apiUrl = "https://eu-api.pdfrest.com"; // EU/GDPR service
const apiKey = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx";
const zugferdPdf = "/path/to/zugferd-invoice.pdf";

async function validateZugferdPdf() {
  const upload = await axios.post(`${apiUrl}/upload`, fs.createReadStream(zugferdPdf), {
    headers: { "Api-Key": apiKey, "Content-Type": "application/pdf", "Content-Filename": path.basename(zugferdPdf) },
    maxBodyLength: Infinity,
  });
  const response = await axios.post(`${apiUrl}/validated-zugferd`, { id: upload.data.files[0].id }, { headers: { "Api-Key": apiKey } });
  console.log(JSON.stringify(response.data, null, 2));
}

validateZugferdPdf().catch((error) => { console.error(error.response?.data || error.message); process.exitCode = 1; });

File: Python/Endpoint Examples/JSON Payload/validated-zugferd.py
import json
import os
import requests

# Upload a hybrid PDF, then validate its ZUGFeRD / Factur-X package by resource ID.
api_url = "https://api.pdfrest.com"
# api_url = "https://eu-api.pdfrest.com"  # EU/GDPR service
api_key = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
zugferd_pdf = "/path/to/zugferd-invoice.pdf"

with open(zugferd_pdf, "rb") as source:
    upload = requests.post(f"{api_url}/upload", data=source, headers={"Api-Key": api_key, "Content-Type": "application/pdf", "Content-Filename": os.path.basename(zugferd_pdf)})
upload.raise_for_status()

response = requests.post(f"{api_url}/validated-zugferd", json={"id": upload.json()["files"][0]["id"]}, headers={"Accept": "application/json", "Api-Key": api_key})
print(json.dumps(response.json(), indent=2))
response.raise_for_status()
